ksqldb-functional-tests/src/test/resources/historical_plans/filter_-_apply_filter_lambda_function_to_int_map/7.0.0_1614088997381/spec.json

```diff
@@ -0,0 +1,244 @@
+    "name" : "apply filter lambda function to int map",
+    "inputs" : [ {
+      "topic" : "test_topic",
+      "key" : "one",
+      "value" : {
+        "map" : {
+          "a" : 15,
+          "abc" : 7
+        }
+      }
+    }, {
+      "topic" : "test_topic",
+      "key" : "one",
+      "value" : {
+        "map" : {
+          "ab" : 6,
+          "abcd" : 15
+        }
+      }
+    }, {
+      "topic" : "test_topic",
+      "key" : "one",
+      "value" : {
+        "map" : {
+          "abcd" : null,
+          "abcde" : 25
+        }
+      }
+    }, {
+      "topic" : "test_topic",
+      "key" : "one",
+      "value" : {
+        "map" : {
+          "null" : 15,
+          "abcde" : 25
+        }
+      }
+    }, {
+      "topic" : "test_topic",
+      "key" : "two",
+      "value" : {
+        "map" : null
+      }
+    } ],
+    "outputs" : [ {
+      "topic" : "OUTPUT",
+      "key" : "one",
+      "value" : {
+        "LAMBDA" : { }
+      }
+    }, {
+      "topic" : "OUTPUT",
+      "key" : "one",
+      "value" : {
+        "LAMBDA" : {
+          "abcd" : 15
+        }
+      }
+    }, {
+      "topic" : "OUTPUT",
+      "key" : "one",
+      "value" : {
+        "LAMBDA" : null
+      }
+    }, {
+      "topic" : "OUTPUT",
+      "key" : "one",
+      "value" : {
+        "LAMBDA" : {
+          "null" : 15,
+          "abcde" : 25
+        }
+      }
+    }, {
+      "topic" : "OUTPUT",
+      "key" : "two",
+      "value" : {
+        "LAMBDA" : null
+      }
+    } ],
+    "topics" : [ {
+      "name" : "OUTPUT",
+      "replicas" : 1,
+      "numPartitions" : 4
+    }, {
+      "name" : "test_topic",
+      "valueSchema" : {
+        "type" : "record",
+        "name" : "KsqlDataSourceSchema",
+        "namespace" : "io.confluent.ksql.avro_schemas",
+        "fields" : [ {
+          "name" : "MAP",
+          "type" : [ "null", {
+            "type" : "array",
+            "items" : {
+              "type" : "record",
+              "name" : "KsqlDataSourceSchema_MAP",
+              "fields" : [ {
+                "name" : "key",
+                "type" : [ "null", "string" ],
+                "default" : null
+              }, {
+                "name" : "value",
+                "type" : [ "null", "int" ],
+                "default" : null
+              } ],
+              "connect.internal.type" : "MapEntry"
+            },
+            "connect.name" : "io.confluent.ksql.avro_schemas.KsqlDataSourceSchema_MAP"
+          } ],
+          "default" : null
+        } ],
+        "connect.name" : "io.confluent.ksql.avro_schemas.KsqlDataSourceSchema"
+      },
+      "valueFormat" : "AVRO",
+      "replicas" : 1,
+      "numPartitions" : 4
+    } ],
+    "statements" : [ "CREATE STREAM test (ID STRING KEY, map MAP<STRING, INTEGER>) WITH (kafka_topic='test_topic', value_format='AVRO');", "CREATE STREAM OUTPUT AS SELECT ID, filter(map, (x,y) => (len(x) * 5 > 15) AND y % 5 = 0) AS lambda FROM test;" ],
```

Comment:
Wondering why use avro for maps and json for lists?

Reply:
Ah, that was just random. I think there's JSON for 1 list and then everything else is AVRO. Do you think it's better to have them more distributed?

Reply:
I think it's fine, just wondering if it is intentional for reasons that I was not aware :)